src/V3Assert.cpp

```diff
@@ -592,30 +658,8 @@ class AssertVisitor final : public VNVisitor {
             ticks = VN_AS(nodep->ticksp(), Const)->toUInt();
         }
         UASSERT_OBJ(ticks >= 1, nodep, "0 tick should have been checked in V3Width");
-        AstNodeExpr* const exprp = nodep->exprp()->unlinkFrBack();
-        AstNodeExpr* inp = newSampledExpr(exprp);
-        AstVar* invarp = nullptr;
-        AstSenTree* const sentreep = nodep->sentreep()->unlinkFrBack();
-        AstAlways* const alwaysp
-            = new AstAlways{nodep->fileline(), VAlwaysKwd::ALWAYS, sentreep, nullptr};
-        m_modp->addStmtsp(alwaysp);
-        for (uint32_t i = 0; i < ticks; ++i) {
-            // TODO recognize AstVarRef is getting delayed and share variables between
-            // $pasts with same reference (or same expression).  Saves downstream
-            // optimizations from identifying and removing duplication.
-            AstVar* const outvarp = new AstVar{
-                nodep->fileline(), VVarType::MODULETEMP,
-                "_Vpast_" + cvtToStr(m_modPastNum++) + "_" + cvtToStr(i), inp->dtypep()};
-            outvarp->lifetime(VLifetime::AUTOMATIC_EXPLICIT);
-            ++m_statPastVars;
-            m_modp->addStmtsp(outvarp);
-            AstNode* const assp = new AstAssignDly{
-                nodep->fileline(), new AstVarRef{nodep->fileline(), outvarp, VAccess::WRITE}, inp};
-            alwaysp->addStmtsp(assp);
-            // UINFOTREE(9, assp, "", "ass");
-            invarp = outvarp;
-            inp = new AstVarRef{nodep->fileline(), invarp, VAccess::READ};
-        }
+        AstNodeExpr* const exprp = newSampledExpr(nodep->exprp()->unlinkFrBack());
```

Comment:
Note here we call `newSampledExpr` on the operand of `$past`, but sometiems AstPast is under AstSampled already, in which case `m_inSampled` is true and then the VarRefs under it will be wrapped in AstSampled already, so we effectively sample a sampled value. This does not seem right to me but I erred on the side of caution and not changed it. This does prevent some reuse in `t_past`.